src/app/modules/main/profile_section/wallet/networks/view.nim

```diff
@@ -42,29 +50,69 @@ QtObject:
     self.areTestNetworksEnabled = not self.areTestNetworksEnabled
     self.areTestNetworksEnabledChanged()
 
-  proc networksChanged*(self: View) {.signal.}
-
-  proc getNetworks(self: View): QVariant {.slot.} =
-    return newQVariant(self.networks)
-
-  QtProperty[QVariant] networks:
-    read = getNetworks
-    notify = networksChanged
-
-  proc load*(self: View, networks: seq[NetworkDto]) =
-    var items: seq[Item] = @[]
-    for n in networks:
-      items.add(initItem(
-        n.chainId,
-        n.layer,
-        n.chainName,
-        n.iconUrl,
-        n.shortName,
-        n.chainColor
-      ))
-
-    self.networks.setItems(items)
+  proc combinedNetworksChanged*(self: View) {.signal.}
+  proc getCombinedNetworks(self: View): QVariant {.slot.} =
+    return newQVariant(self.combinedNetworks)
+  QtProperty[QVariant] combinedNetworks:
+    read = getCombinedNetworks
+    notify = combinedNetworksChanged
+
+  proc layer1NetworksChanged*(self: View) {.signal.}
+  proc getLayer1Networks(self: View): QVariant {.slot.} =
+    return newQVariant(self.layer1Networks)
+  QtProperty[QVariant] layer1Networks:
+    read = getLayer1Networks
+    notify = layer1NetworksChanged
+
+  proc layer2NetworksChanged*(self: View) {.signal.}
+  proc getLayer2Networks(self: View): QVariant {.slot.} =
+    return newQVariant(self.layer2Networks)
+  QtProperty[QVariant] layer2Networks:
+    read = getLayer2Networks
+    notify = layer2NetworksChanged
+
+  proc load*(self: View, combinedNetworks: seq[CombinedNetworkDto]) =
+    var combinedItems: seq[CombinedItem] = @[]
+    var layer1Items: seq[CombinedItem] = @[]
+    var layer2Items: seq[CombinedItem] = @[]
+    for n in combinedNetworks:
+      var prod = newItem(
+          n.prod.chainId,
+          n.prod.layer,
+          n.prod.chainName,
+          n.prod.iconUrl,
+          n.prod.shortName,
+          n.prod.chainColor,
+          n.prod.rpcURL,
+          n.prod.fallbackURL,
+          n.prod.blockExplorerURL,
+          n.prod.nativeCurrencySymbol
+        )
+      var test = newItem(
+          n.test.chainId,
+          n.test.layer,
+          n.test.chainName,
+          n.test.iconUrl,
+          n.test.shortName,
+          n.test.chainColor,
+          n.test.rpcURL,
+          n.test.fallbackURL,
+          n.test.blockExplorerURL,
+          n.test.nativeCurrencySymbol
+        )
+      combinedItems.add(initCombinedItem(prod,test))
+      if n.prod.layer == 1:
+        layer1Items.add(initCombinedItem(prod,test))
+      else:
+        layer2Items.add(initCombinedItem(prod,test))
+
+    self.combinedNetworks.setItems(combinedItems)
+    self.layer1Networks.setItems(layer1Items)
+    self.layer2Networks.setItems(layer2Items)
```

Comment:
This is just violating the DRY principle; you can easily filter those using e.g. `SortFilterProxyModel` and that boolean I mentioned above (or any other criteria in fact)